x/crosschain/keeper/abci.go

```diff
@@ -85,6 +84,23 @@ func (k Keeper) CheckAndUpdateCctxGasPrice(
 	}
 	gasPriceIncrease := medianGasPrice.MulUint64(uint64(flags.GasPriceIncreasePercent)).QuoUint64(100)
 
+	// compute new gas price
+	currentGasPrice, err := cctx.GetCurrentOutTxParam().GetGasPrice()
+	if err != nil {
+		return math.ZeroUint(), math.ZeroUint(), err
+	}
+	newGasPrice := math.NewUint(currentGasPrice).Add(gasPriceIncrease)
```

Comment:
Is it possible that this `currentGasPrice` is outdated and significantly deviates (higher or lower) from the `medianGasPrice`? Are we `adjusting` the "gasPrice" based on (by percentage) the `currentGasPrice` or directly based on the dynamic `medianGasPrice`?

Reply:
`currentGasPrice` can be higher than `medianGasPrice` if it has already been increased by the same method previously. The gas price increase is calculated from the median gas price.